src/service/resources-impl.js

```diff
@@ -458,6 +477,9 @@ export class ResourcesImpl {
    * @return {boolean}
    */
   isUnderBuildQuota_() {
+    if (this.buildInChunks_ && this.buildsThisPass_ >= MAX_BUILD_CHUNK_SIZE) {
```

Comment:
Why not rely on chunking limiting the amount of work based on actual time instead?